### DISCUSSION TITLE
Write additional header items into the Well information section

### DISCUSSION THREAD

**Discussion by Taisi** [kinverarity1/lasio#560] [Q&A] [answered]
Dear lasio developers, 
first of all thanks a lot for this wonderful package!

I was wondering whether it is possible to write additional header items into a las-file when creating it from scratch. 
I  have to explicitly provide X and Y coordinates. 

Assigning  

```
las = lasio.LASFile()
header.well.XCOORD = 123456
```
Results in 
```
In[]: las.well.XCOORD
Out[]: 123456
```

However it doesn't apper in the resulting file if I use
`las.write(file, version = 2)`

Maybe there are some other parameters? 

Thanks a lot in advance!

**Comment by kinverarity1** [comment]
Certainly it should be possible! I think try 

las.well['XCOORD'] = 123456

**Comment by Taisi** [comment]
Thank a lot! 
trying it directly raises an KeyError
raise KeyError("%s not in %s" % (key, self.keys()))

KeyError: "XCOORD not in ['STRT', 'STOP', 'STEP', 'NULL', 'COMP', 'WELL', 'FLD', 'LOC', 'PROV', 'CNTY', 'STAT', 'CTRY', 'SRVC', 'DATE', 'UWI', 'API']"

But I'm going through the las_items.py now and may find a way.

**Comment by kinverarity1** [comment, marked as answer]
Ah, you need to use lasio.HeaderItem

https://lasio.readthedocs.io/en/latest/header-section.html

@Taisi Try:

```python
las.well.XCOORD= lasio.HeaderItem(mnemonic='XCOORD', value=123456)
```

**Comment by dcslagel** [comment]
Is this something that could/should have a clearer example (and/or explanation) in the [Building a LAS file from scratch](https://lasio.readthedocs.io/en/latest/building.html) part of the documentation?

**Reply by kinverarity1** [reply]
Yep I think probably - there is an example here:

```las.params.DRILL = lasio.HeaderItem(mnemonic='DRILL', value='John Smith', descr='Driller on site')```

But it wouldn't hurt to add that to the link you copied either!

**Reply by Taisi** [reply]
For me it was not clear how to directly access the list of header items to add a new one, but the advise to use lasio.HeaderItem was enough. 

And the fact that 

header.well.XCOORD = 123456 does not raise any error, although the XCCORD was not specified in advance was a bit confusing. 

I guess an example can do now harm. 
Thanks a lot for your help!

**Comment by Taisi** [comment]
Thanks a lot!

I did it this way and it works perfectly fine!
```
las = lio.LASFile()
header= las.header
well_section = header['Well']
well_section.append(HeaderItem(mnemonic="XCOORD", unit="", value="", descr="XCOORD"))
well_section["XCOORD"] = X
```
